player/video.c

```diff
@@ -1129,6 +1129,34 @@ void write_video(struct MPContext *mpctx)
 
     // Filter output is different from VO input?
     struct mp_image_params *p = &mpctx->next_frames[0]->params;
+    // Inject vo crop to notify and reconfig if needed
+    struct mp_rect rc;
+    m_rect_apply(&rc, p->w, p->h, &vo->opts->video_crop);
+    if (rc.x1 > 0 && rc.y1 > 0)
+    {
+        p->crop = rc;
+        if (!mp_image_crop_valid(p)) {
+            char *rc = m_option_type_rect.print(NULL, &vo->opts->video_crop);
```

Comment:
This seems to cause a shadowing warning as the exact same variable name is utilized for both the `mp_rect` as well as the print-out of the invalid rectangle definition.
```
../player/video.c: In function 'write_video':
../player/video.c:1139:19: warning: declaration of 'rc' shadows a previous local [-Wshadow]
 1139 |             char *rc = m_option_type_rect.print(NULL, &vo->opts->video_crop);
      |                   ^~
../player/video.c:1133:20: note: shadowed declaration is here
 1133 |     struct mp_rect rc;
      |                    ^~
```

Reply:
fixed